File: pos2angle.m
function [phi1, phi2] = pos2angle(dx, dy)
    if (sqrt(dx^2 + dy^2) > 250)
        % modify to return previous angles
        return;
    end
    la = 250;
    beta = acos(1-(dx^2+dy^2)/(2*la^2))
    delta = acos(sqrt(dx^2+dy^2)/(2*la));
    gamma = atan2(dy,dx);
    alpha = delta + gamma

    phi1 = pi / 2 - alpha;
    phi2 = pi * 3 / 2 - (alpha + beta);

    if (phi2 > pi)
        phi2 = phi2 - pi
    end

    if (phi1 > pi)
        phi1 = phi1 - pi
    end

    phi1deg = phi1 * 180 /pi
    phi2deg = phi2 * 180 / pi
    
    
end

File: pos2angle.c
#include <math.h>

 double * pos2angle(int dx, int dy) { 
    static double angles[2];
    // if position is impossible to reach, return previous values
    if (sqrt(dx^2 + dy^2) > 250) return angles;

    int la = 250;
    double beta = acos(1-(dx^2+dy^2)/(2*la^2));
    double delta = acos(sqrt(dx^2+dy^2)/(2*la));
    double gamma = atan2(dy,dx);
    double alpha = delta + gamma;

    double phi1 = pi / 2 - alpha;
    double phi2 = pi * 3 / 2 - (alpha + beta);

    // in case angles over 180 degrees is returned
    if (phi2 > pi)  phi2 = phi2 - pi;
    if (phi1 > pi) phi1 = phi1 - pi;
    if (phi2 < pi)  phi2 = phi2 + pi;
    if (phi1 < pi) phi1 = phi1 + pi;

    angles[0] = phi1;
    angles[1] = phi2;

    return angles;

    // phi1deg = phi1 * 180 /pi
    // phi2deg = phi2 * 180 / pi
 }
